drivers/mipi_dbi/mipi_dbi_bflb.c

```diff
@@ -0,0 +1,773 @@
+#define DBI_FIFO_FORMAT_RnBGR	4U
+#define DBI_FIFO_FORMAT_BnRGB	5U
+
+#define DBI_MAX_CMD_WRITE_LEN	256
+
+struct mipi_dbi_bflb_config {
+	uint32_t base;
+	const struct pinctrl_dev_config *pincfg;
+	void (*irq_config_func)(const struct device *dev);
+	const struct gpio_dt_spec reset_gpio;
+	const struct gpio_dt_spec dc_gpio;
+	const struct gpio_dt_spec *cs_gpios;
+	size_t num_cs_gpios;
+};
+
+struct mipi_dbi_bflb_dma_data {
+	const struct device *dev;
+	int channel;
+	struct dma_config config;
+	struct dma_block_config block;
+	uint8_t	*buf;
+	size_t	len;
+};
+
+struct mipi_dbi_bflb_data {
+	struct mipi_dbi_config configured;
+	struct k_sem lock;
+	struct mipi_dbi_bflb_dma_data dma;
+	bool release_at_dma_end;
+};
+
+static inline bool mipi_dbi_has_pin(const struct gpio_dt_spec *spec)
+{
+	return spec->port != NULL;
+}
+
+static inline void mipi_dbi_bflb_cs_control(const struct device *dev,
+					    const struct mipi_dbi_config *config,
+					    bool on, bool force_off)
+{
+	const struct mipi_dbi_bflb_config *cfg = dev->config;
+
+	if (config->config.slave < cfg->num_cs_gpios) {
+		if (mipi_dbi_has_pin(&cfg->cs_gpios[config->config.slave])) {
+			if (on) {
+				gpio_pin_set_dt(&cfg->cs_gpios[config->config.slave], 1);
+				k_busy_wait(config->config.cs.delay);
+			} else {
+				if (!force_off
+				    && (config->config.operation & SPI_HOLD_ON_CS
+					|| IS_ENABLED(CONFIG_MIPI_DBI_BFLB_ASSUME_HOLD_ON_CS))) {
+					return;
+				}
+				k_busy_wait(config->config.cs.delay);
+				gpio_pin_set_dt(&cfg->cs_gpios[config->config.slave], 0);
+			}
+		}
+	}
+}
+
+static bool mipi_dbi_bflb_bus_busy(const struct device *dev)
+{
+	const struct mipi_dbi_bflb_config *config = dev->config;
+	volatile uint32_t tmp;
+
+	tmp = sys_read32(config->base + DBI_CONFIG_OFFSET);
+
+	return (tmp & DBI_STS_DBI_BUS_BUSY) != 0;
+}
+
+static void mipi_dbi_bflb_clear_fifo(const struct device *dev)
+{
+	const struct mipi_dbi_bflb_config *config = dev->config;
+	volatile uint32_t tmp;
+
+	tmp = sys_read32(config->base + DBI_FIFO_CONFIG_0_OFFSET);
+	tmp |= DBI_TX_FIFO_CLR;
+	sys_write32(tmp, config->base + DBI_FIFO_CONFIG_0_OFFSET);
+}
+
+static int mipi_dbi_bflb_trigger(const struct device *dev)
+{
+	volatile uint32_t tmp;
+	const struct mipi_dbi_bflb_config *config = dev->config;
+	k_timepoint_t end_timeout =
+		sys_timepoint_calc(K_MSEC(DBI_WAIT_TIMEOUT_MS));
+
+	while (mipi_dbi_bflb_bus_busy(dev) && !sys_timepoint_expired(end_timeout)) {
+	}
+	if (sys_timepoint_expired(end_timeout)) {
+		return -ETIMEDOUT;
+	}
+
+	tmp = sys_read32(config->base + DBI_CONFIG_OFFSET);
+	tmp |= DBI_CR_DBI_EN;
+	sys_write32(tmp, config->base + DBI_CONFIG_OFFSET);
+
+	return 0;
+}
+
+static int mipi_dbi_bflb_detrigger(const struct device *dev)
+{
+	volatile uint32_t tmp;
+	const struct mipi_dbi_bflb_config *config = dev->config;
+	k_timepoint_t end_timeout =
+		sys_timepoint_calc(K_MSEC(DBI_WAIT_TIMEOUT_MS));
+
+	while (mipi_dbi_bflb_bus_busy(dev) && !sys_timepoint_expired(end_timeout)) {
+	}
```

Comment:
If you wanted a `k_sleep()` above, probably here too

Reply:
done